src/App.tsx

```diff
@@ -15,33 +16,112 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setIsStarted] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  function getReorderedGoods(goods: string[]) {
+    const visibleGoods = [...goods];
+
+    switch (sortType) {
+      case SortType.ALPABET:
+        visibleGoods.sort((goodA, goodB) => goodA.localeCompare(goodB));
+        break;
+
+      case SortType.LENGTH:
+        visibleGoods.sort((goodA, goodB) => goodA.length - goodB.length);
+        break;
+
+      default:
+        break;
+    }
+
+    return isReversed ? visibleGoods.reverse() : visibleGoods;
+  }
+
+  function reverse() {
+    if (isReversed) {
+      setIsReversed(false);
+    } else {
+      setIsReversed(true);
+    }
+  }
+
+  return (
+    <article className="App panel is-warning">
+      {!isStarted && (
+        <button
+          className="button is-large is-info is-fullwidth"
+          type="button"
+          onClick={() => setIsStarted(true)}
+        >
+          Start
+        </button>
+      )}
+
+      {isStarted && (
+        <>
+          <p className="App__title panel-heading">
+            Shoping-list
+          </p>
+
+          <div className="panel-tabs">
+            <button
+              className="button is-white"
+              type="button"
+              onClick={() => setSortType(SortType.ALPABET)}
+            >
+              Sort alphabetically
+            </button>
+
+            <button
+              className="button is-white"
+              type="button"
+              onClick={() => setSortType(SortType.LENGTH)}
+            >
+              Sort by length
+            </button>
+
+            <button
+              className="button is-white"
+              type="button"
+              onClick={() => reverse()}
+            >
+              Reverse
+            </button>
+
+            <button
+              className="button is-danger is-light"
+              type="button"
+              onClick={() => {
+                setSortType(SortType.NONE);
+                setIsReversed(false);
+              }}
```

Comment:
move this logic to function-handlers, it's not a common practice to embed code directly